pkg/engine/runtime/terraform_runtime.go

```diff
@@ -0,0 +1,140 @@
+package runtime
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/imdario/mergo"
+	"github.com/spf13/afero"
+	"kusionstack.io/kusion/pkg/engine/models"
+	"kusionstack.io/kusion/pkg/status"
+	"kusionstack.io/kusion/pkg/terraform"
+)
+
+var _ Runtime = &TerraformRuntime{}
+
+type TerraformRuntime struct {
+	terraform.WorkspaceStore
+}
+
+func NewTerraformRuntime() (Runtime, error) {
+	fs := afero.Afero{Fs: afero.NewOsFs()}
+	ws, err := terraform.GetWorkspaceStore(fs)
+	if err != nil {
+		return nil, err
+	}
+	TFRuntime := &TerraformRuntime{ws}
+	return TFRuntime, nil
+}
+
+// Apply terraform apply resource
+func (t *TerraformRuntime) Apply(ctx context.Context, request *ApplyRequest) *ApplyResponse {
+	planState := request.PlanResource
+	w, ok := t.Store[planState.ResourceKey()]
+	if !ok {
+		err := t.Create(ctx, planState)
+		if err != nil {
+			return &ApplyResponse{nil, status.NewErrorStatus(err)}
+		}
+		w = t.Store[planState.ResourceKey()]
+	}
+
+	// get terraform provider version
+	providerAddr, err := w.GetProvider()
+	if err != nil {
+		return &ApplyResponse{nil, status.NewErrorStatus(err)}
+	}
+
+	// terraform dry run merge state
+	if request.DryRun {
```

Comment:
This is a temporary logic to implement dryRun in tf runtime. Please add //todo comments for this logic